src/util/token.js

```diff
@@ -18,9 +18,11 @@ module.exports = {
     const seedBuffer = base64url.toBuffer(seed)
     const publicKeyBuffer = base64url.toBuffer(publicKey)
 
-    return base64url(sodium.crypto_scalarmult(
+    const shared = sodium.crypto_scalarmult(
       sodium.crypto_hash_sha256(seedBuffer),
       publicKeyBuffer
-    ))
+    )
+
+    return base64url(sodium.crypto_auth_hmacsha256(base64url.toBuffer('token'), shared))
```

Comment:
Seems pretty weird to me to interpret the string as base64 instead of ASCII. I would replace `base64url.toBuffer('token')` with `Buffer.from('token', 'ascii')`.